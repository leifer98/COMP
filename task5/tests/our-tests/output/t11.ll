@.str0 = constant [23 x i8] c"Error division by zero\00"
declare i32 @scanf(i8*, ...)
declare i32 @printf(i8*, ...)
declare void @exit(i32)
@.int_specifier_scan = constant [3 x i8] c"%d\00"
@.int_specifier = constant [4 x i8] c"%d\0A\00"
@.str_specifier = constant [4 x i8] c"%s\0A\00"
define i32 @readi(i32) {
%ret_val = alloca i32
%spec_ptr = getelementptr [3 x i8], [3 x i8]* @.int_specifier_scan, i32 0, i32 0
call i32 (i8*, ...) @scanf(i8* %spec_ptr, i32* %ret_val)
%val = load i32, i32* %ret_val
ret i32 %val
}
define void @printi(i32) {
%spec_ptr = getelementptr [4 x i8], [4 x i8]* @.int_specifier, i32 0, i32 0
call i32 (i8*, ...) @printf(i8* %spec_ptr, i32 %0)
ret void
}
define void @print(i8*) {
%spec_ptr = getelementptr [4 x i8], [4 x i8]* @.str_specifier, i32 0, i32 0
call i32 (i8*, ...) @printf(i8* %spec_ptr, i8* %0)
ret void
}
define void @main() {
%t1 = add i32 0, 50
%t2 = call i32 @getZero()
%t3 = icmp eq i32 %t2, 0
br i1 %t3, label %label_0, label %label_1
label_0:
%t4 = getelementptr inbounds [23 x i8], [23 x i8]* @.str0, i32 0, i32 0
call void @print(i8* %t4)
call void @exit(i32 1)
br label %label_1
label_1:
%t0 = udiv i32 %t1, %t2
%t5 = alloca i32 
store i32 %t0, i32* %t5
ret void
}
define i32 @getZero() {
%t7 = add i32 0, 2
%t8 = add i32 0, 2
%t6 = sub i32 %t7, %t8
ret i32 %t6
}
