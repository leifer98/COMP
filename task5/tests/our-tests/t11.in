void main() {
    int result = 50 / getZero();
}

int getZero() {
    return 2 - 2; // Always returns 0
}
