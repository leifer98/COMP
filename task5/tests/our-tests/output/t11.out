PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.
Stack dump:
0.	Program arguments: lli
 #0 0x00007f58dc2b2d01 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0xe3fd01)
 #1 0x00007f58dc2b0a3e llvm::sys::RunSignalHandlers() (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0xe3da3e)
 #2 0x00007f58dc2b3236 (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0xe40236)
 #3 0x00007f58daf59520 (/lib/x86_64-linux-gnu/libc.so.6+0x42520)
 #4 0x00007f58e1d8a086 
 #5 0x00007f58ddcfa84d llvm::orc::runAsMain(int (*)(int, char**), llvm::ArrayRef<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, llvm::Optional<llvm::StringRef>) (/lib/x86_64-linux-gnu/libLLVM-14.so.1+0x288784d)
 #6 0x0000000000418110 runOrcJIT(char const*) (/usr/lib/llvm-14/bin/lli+0x418110)
 #7 0x0000000000412cc8 main (/usr/lib/llvm-14/bin/lli+0x412cc8)
 #8 0x00007f58daf40d90 __libc_start_call_main ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
 #9 0x00007f58daf40e40 call_init ./csu/../csu/libc-start.c:128:20
#10 0x00007f58daf40e40 __libc_start_main ./csu/../csu/libc-start.c:379:5
#11 0x000000000040fe15 _start (/usr/lib/llvm-14/bin/lli+0x40fe15)
